pkg/platform/runtime/setup/implementations/alternative/artifact.go

```diff
@@ -23,7 +23,23 @@ func NewArtifactSetup(artifactID artifact.ArtifactID, store *store.Store) *Artif
 	return &ArtifactSetup{artifactID, store}
 }
 
-func (as *ArtifactSetup) EnvDef(tmpDir string) (*envdef.EnvironmentDefinition, error) {
+func (as *ArtifactSetup) PrepareEnvDef(tmpDir, installDir string, constants envdef.Constants) (*envdef.EnvironmentDefinition, error) {
+	// Source the environment definition from the extracted archive
+	ed, err := as.newEnvDef(tmpDir)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not load environment definitions for artifact.")
+	}
+
+	// Expand the environment definition variables
+	ed = ed.ExpandVariables(constants)
+
+	// Ensure that the replacement values are set to the correct directory
+	ed.ReplacePath(filepath.Join(tmpDir, ed.InstallDir), installDir)
```

Comment:
I understand there's a chicken and egg problem here. Can you add a comment explaining why this is necessary this way? Looking at the code my first thought is "why use a temp dir here at all?".